### DISCUSSION TITLE
Ignore keyboard shortcuts while TextEdit is active?

### DISCUSSION THREAD

**Discussion by tom-kaufman** [emilk/egui#3469] [Q&A] [answered]
I have a lot of shortcuts in my app that would collide with normal data entry into a TextEdit. How can I capture the active state of my `TextEdit`s, so that I can ignore checking for shortcuts while they're active?

**Comment by tom-kaufman** [comment, marked as answer]
I figured this out. I changed all my `TextEdit`s to use `.show(ui)`, and extracted the `TextEditOutput`, to check if `.cursor_range.is_some()`. My app now holds a bool, that I `or` all of these values together to check if any `TextEdit` are active. I reset the bool to `false` on every call of `update()`, and put my shortcut handling behind an `if` statement at the end of `update()`.

**Comment by thomasa88** [comment]
You can use [Response::has_focus()](https://docs.rs/egui/latest/egui/response/struct.Response.html#method.has_focus) (e.g. from ui.add()).
